### ISSUE TITLE
Exposing intermediate features from a Backbone

### ISSUE THREAD

**Issue by AffineParameter** [jalabort/nucleus#30]
https://github.com/jalabort/nucleus/blob/93cbb3d0b1e1b697fbd7e9f2e0fa9f2eca5ad202/nucleus/detection/backbones/managers.py#L81-L147

In some circumstances, I think it would be useful to pass lower-level (higher-resolution) features to a head network through `keras.Model`'s `outputs` kwarg. Selecting the ones you want to expose could probably use something like the current `trainable` kwarg pattern. 

I don't think we would need to worry about gradient control here, as the head can be configured to pass gradients through to the lower-level weights if needed or not.
